src/commands/moderators/micMuteAppeal.ts

```diff
@@ -0,0 +1,76 @@
+import { GuildMember } from 'discord.js'
+
+import { Environment } from '../../config.js'
+import { addUserModerationLogEntry } from '../../features/profileInspector/index.js'
+import { UserModerationLogEntryType } from '../../types/UserModerationLogType.js'
+import { defineCommandHandler } from '../../types/defineCommandHandler.js'
+
+export default defineCommandHandler({
+  data: {
+    name: 'appeal-for-server-mute',
+    description:
+      'Appeal for microphone muted. Use when server muted only, else you will be timed out for one minute.',
+  },
+  ephemeral: true,
+  execute: async (botContext, interaction) => {
+    if (
+      !interaction.isChatInputCommand() ||
+      interaction.channelId !== Environment.MIC_MUTE_APPEAL_CHANNEL_ID
+    ) {
+      interaction.deleteReply()
+      return
+    }
+    if (interaction.member instanceof GuildMember) {
+      //when start the bot, all user voice state might be null.This if statement is to prevent it.
+      if (interaction.member.voice.serverMute === null) {
+        interaction.editReply('please join voice channel')
+        return
+      }
+      //prevent spamming appeal when the user is not mute
+      if (interaction.member.voice.serverMute === false) {
+        await interaction.editReply(
+          'you are not muted, you will be timed out for one minute due to the false mute appeal.',
+        )
+        setTimeout(async () => {
+          await interaction.deleteReply()
+        }, 5000)
+        try {
+          //time out does not work on user with higher role hierachy.
+          await interaction.member.timeout(1000 * 60)
+        } catch (err) {
+          
+        }
+        return
+      }
+      //if the user is mute, unmute the user.
+      //unmuting might be depended on reason why user is server muted.
+      else {
+        try {
+          if (true) {
```

Comment:
Can you extract it to function can call that function here?

Reply:
can you clarify your statement ,i dont get what you desire.

Reply:
Anyway, I commit based on what in understand. (i understand that you desire me to change if(true) to if(someToDoFunction).)

Reply:
I have created it as TODO function.
https://github.com/kaogeek/kaogeek-discord-bot/blob/b882a891e456a6c64744760f1778f04e7fecf40f/src/commands/moderators/micMuteAppeal.ts#L46
https://github.com/kaogeek/kaogeek-discord-bot/blob/b882a891e456a6c64744760f1778f04e7fecf40f/src/commands/moderators/micMuteAppeal.ts#L74-L80
I might implement it along with moderationlog metadata of muting user.